### PR TITLE
Add "Naming convention" to `test_cases/README.md`

### PR DESCRIPTION
**Pull request by sobolevn**
Refs https://github.com/python/typeshed/pull/8520

### CODE REVIEW

File: test_cases/README.md [hunk 1]
```diff
@@ -52,6 +52,15 @@ annotations,
 [this page](https://typing.readthedocs.io/en/latest/source/quality.html#testing-using-assert-type-and-warn-unused-ignores)
 provides a useful guide.
 
+### Naming convention
+
+Use the same top-level name for the module / package you would like to test.
+Use `test_${thing}.py` naming pattern for individual test files.
+For example:
+- Tests for `gather` defined in `stdlib/asyncio/tasks.pyi` will go into `stdlib/asyncio/test_gather.py`
```
Comment (JelleZijlstra) [inline test_cases/README.md:60]:
```suggestion
- Tests for `gather` defined in `stdlib/asyncio/tasks.pyi` will go into `stdlib/asyncio/tasks/test_gather.py`
```

File: test_cases/README.md [hunk 2]
```diff
@@ -52,6 +52,15 @@ annotations,
...
+- Tests for `ExitStack` defined in `stdlib/contextlib.pyi` will go into `stdlib/test_contextlib.py`
+- Tests for `dict` defined in `stdlib/builtins.pyi` will go into `stdlib/stdlib/test_dict.py`
```
Comment (JelleZijlstra) [inline test_cases/README.md:62]:
```suggestion
- Tests for `dict` defined in `stdlib/builtins.pyi` will go into `stdlib/builtins/test_dict.py`
```

File: test_cases/README.md [hunk 3]
```diff
@@ -52,6 +52,15 @@ annotations,
 [this page](https://typing.readthedocs.io/en/latest/source/quality.html#testing-using-assert-type-and-warn-unused-ignores)
 provides a useful guide.
 
+### Naming convention
+
+Use the same top-level name for the module / package you would like to test.
+Use `test_${thing}.py` naming pattern for individual test files.
+For example:
+- Tests for `gather` defined in `stdlib/asyncio/tasks.pyi` will go into `stdlib/asyncio/test_gather.py`
+- Tests for `ExitStack` defined in `stdlib/contextlib.pyi` will go into `stdlib/test_contextlib.py`
```
Comment (JelleZijlstra) [inline test_cases/README.md:61]: These examples don't show why ExitStack goes into `test_contextlib` directly, but `pow` doesn't go into `test_builtins`. We should add something like "By default, tests go into a test file with the same name as the stub file, prefixed with `test_`. For example: test_contextlib. If that file becomes too big, we instead create a directory with files named after individual objects being tested. For example: builtins/test_dict."

**Review by JelleZijlstra — CHANGES_REQUESTED** [review]

File: test_cases/README.md (hunk 1 above)
Comment (sobolevn) [reply test_cases/README.md:60]: Should we move it then? https://github.com/python/typeshed/blob/master/test_cases/stdlib/asyncio/test_gather.py

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**